### PR TITLE
Next generation

### LINKED ISSUE #14: Multilingual site support
Hi Vicky, i really like your theme, nice work! I was wondering if you were planning on adding support for multilingual sites.

As far as I can see, hugo supports translation of strings so it should be easy enough to move all untranslated text in templates to a data file, make sure the templates are using relLangURL, absLangURL and displaying some sort of link to translated content (posts, etc).

If you want I can give you a hand implementing the missing functionality in the theme and submit a pull request.

### LINKED ISSUE #29: Rework theme css to be more organized/not include Bulma
I'll admit that since creating this theme, I've learned more about how to better organize CSS (Sass files specifically) and also learned a few lessons about dependencies.

I'd like to rewrite the theme's CSS (classes in the HTML files, and the Sass/CSS files themselves) to be more DRY, and independent of another framework (no Bulma). My preference would be to use Flexbox. 

I don't currently have the time to do this so I'm opening it up as an invitation to contribute. This is an issue best suited to someone with good working knowledge of CSS, Sass, and Flexbox.

The idea is to replicate the current design and two accompanying color versions (light and dark) with DRY code, using Flexbox. The CSS for each of the two color versions could easily be created using variables that can be changed in one Sass file.

### LINKED ISSUE #46: Background scrolling on mobile phone
After opening a modal window e. g. after clicking on "Design", and scrolling the page up or down, the (nearly invisible) main page in the background is scrolling. After closing the modal by clicking the X in the upper right, the main page may be on a totally different position than before opening the modal.

### LINKED ISSUE #54: Remove enforce_ssl setting
Enforcing ssl/tls is nothing what should be done via js. A webserver should take care of this, not hugo. I propose to remove this option.

@vickylai Thoughts?

### LINKED ISSUE #63: Configure about and contacts text without creating a new page
Currently to configure the about and contacts text, the user has to create a seperate page for both of them. But this also means that this pages are render with there own URL.

I propose to move `content/contact.md` text to a param and `content/about.md` to `content/index.md`. This would of course break backwards compatibility, but we could do this in a major release. I think, there will be no features lost.

### LINKED ISSUE #64: Cleanup config.toml
I my opinion `exampleSite/config.toml`is quite overloaded. I propose to remove or relocate some params:

### Code
- `pygmentsStyle`: Is set to `monokai` by default. No need to specify this. See https://gohugo.io/getting-started/configuration/
- `pygmentsCodefences`: if a user really wants this, he/she can enable it my her/his self
- `cacheBuster`: I don`t see the need for this, but I might be wrong.

### Home page configuration
There are a couple of parameters to configure the home page, like `firstName`. `tagLine` and `introHeight`. We could move this to font matter in `content/index.md`.

### Time
I really don\`t see the point in showing users the current time. We are not living in a world where response times for e-mails matter so much, that we need to show when to expect an answer. I propose to remove `localTime`, `timeZone`, `dateForm` and `dateFormFull`.

Just like #63 this breaks backwards compatibility.

### LINKED ISSUE #67: Update Bulma
We currently use a quite old version of bulma (<0.4). This should be upgraded to make use of some HTML5 elements.

### LINKED ISSUE #69: Images on single project pages
Hi, love the theme so far. 

In my first attempt to use this theme, images included in project pages (stored in the `static` directory) render on the main site page and on the `/projects` page, but not on the single pages for individual projects. 

I noticed in layout partials for the main page and project page that these images are referenced using `relURL`, but in the `single.html` file the `relURL` bit is not included. When I add it to my own version of `single.html`, the images load on the single page as they're supposed to.

I can suggest the edit but (seeing as I'm not skilled with Hugo) I'm wondering if this isn't an ideal approach of if I'm overlooking something. 

Thanks.

### PR DESCRIPTION
**Pull request by Hanzei**
This is mostly a complete rewrite of the theme. I tried to make the theme more dry by using [base templates](https://gohugo.io/templates/base), but ended up rearranging most of the code. Sorry @vickylai for this huge PR but I think its easier to review it like this. And it creates less merge conflicts.

#### Added
- [Multilingual support](https://gohugo.io/content-management/multilingual/) (Fixes #14, closes #40, closes #45)
- [Image Processing](https://gohugo.io/content-management/image-processing/)
- [Carousel](https://github.com/OwlCarousel2/OwlCarousel2/) for multiple images in projects
- [Custom Menus](https://gohugo.io/content-management/menus/)
- [Open Graph metadata](http://ogp.me/)
- [Headless bundle](https://gohugo.io/content-management/page-bundles/#headless-bundle) for home page (Fixes #63)
- Test rendered html code with [html-proofer](https://github.com/gjtorikian/html-proofer)
- Added `showRSSButton` option
- Close modals with ESC key

#### Fixed
- A lot of inconsistencies with style
- Background scrolling while modal is open (Fixes #46, closes #62)
- Fix missing images on `projects` single page (Fixes #69)

#### Changed
- Use standard mobile nav on home page
- Update dependencies (Fixes #67)
- Rewrite sass files to be more DRY (Fixes #29)
- Rewrite config file (Fixes #64)

#### Removed
- Remove `enforce_ssl` setting (Fixes #54, closes #56)
- Remove `showBlog` and `showBlog` settings. The sections are now shown if they exist.

## To Do
- [x] Finish and check Spanish translation
- [x] Some assets don't load if baseurl includes a path, eg. https://example.org/site/
- [x] The owlCarousel should be implemented as well for the modal project view
- [x] Fix to big margin for footer top icon
- [x] Update to hugo 0.45
- [x] Dokumentation
- [ ] Update `CHANGELOG.md`
- [ ] More example Images. @vickylai Can you help with this?

@vickylai Please give this a first round of review

### CODE REVIEW

**Comment by pmviva** [comment]
Sure, @Hanzei I'll review the spanish translation this week.

File: exampleSite/content/es/blog/_index.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ---
 title: "Blog"
 ---
+
+Here is some introduction text for my blog
```
Comment (pmviva) [inline exampleSite/content/es/blog/_index.md]: This should be:
"Este es un texto introductorio para mi blog."

File: exampleSite/content/es/projects/_index.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
 ---
 title: "Proyectos"
 ---
+
+Here is some introduction text for my projects
```
Comment (pmviva) [inline exampleSite/content/es/projects/_index.md]: This should be:
"Este es un texto introductorio para mis proyectos."

File: exampleSite/content/es/blog/_index.md (hunk 1 above)
Comment (pmviva) [inline exampleSite/content/es/blog/_index.md]: This should be:
"Este es un texto introductorio para mi blog."

File: i18n/es.toml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    other = "Volver a {{ .Title }}"
+[nav_main]
+    other = "Principal"
+[blog_readingTime]
+    one = "{{ .Count }} minuto de lectura"
+    other = "{{ .Count }} minutos de lectura"
+[blog_shareThis]
+    other = "Comparte esto"
+[footer_text]
+    other = "Tema [**Introduction**](https://github.com/vickylai/hugo-theme-introduction/) para [Hugo](http://gohugo.io/). Made with [<i class=\"fa fa-heart\"></i> and <i class=\"fa fa-coffee\"></i>](https://vickylai.com) by open source contributors."
```
Comment (Hanzei) [inline i18n/es.toml]: @pmviva Could you help me with this one too?

Comment (pmviva) [reply i18n/es.toml]: The first translation is OK, the second one should be:

+[footer_text]
+    other = "Tema [**Introduction**](https://github.com/vickylai/hugo-theme-introduction/) para [Hugo](http://gohugo.io/). Hecho con [<i class=\"fa fa-heart\"></i> y <i class=\"fa fa-coffee\"></i>](https://vickylai.com) por colaboradores de código abierto."

**Comment by vickylai** [comment]
This is very good work. Well done @Hanzei and @pmviva !

__Headless:__

Let's go ahead with implementing headless bundles as in #63. The user can choose to have the pages accessible at a permalink by setting `headless = false` so I am fine with this.

__Multilingual:__

* The translation for the sections and page names I think is still not handled:
  * `index.html` use of `.Site.GetPage` by page name
  * `nav.html` use of `.Site.GetPage` by page name, use of `.Page.Section` and `.Page.Kind`

__Misc:__

* "Fix to big margin for footer top icon" I don't understand?
* The owlCarousel should be implemented as well for the modal project view
* The current timezone in index page Contact section doesn't seem to work. As in #64 I'm okay with taking a vote to see if anyone actually uses this feature and wants it to remain.
* Should `exampleSite/resources/_gen/` be included or ignored?

**Comment by Hanzei** [comment]
> Let's go ahead with implementing headless bundles as in #63. The user can choose to have the pages accessible at a permalink by setting headless = false so I am fine with this.

Done, see 0f2d3808ee3b548f1861c602489432015465f71f

> - The translation for the sections and page names I think is still not handled:
>     - index.html use of .Site.GetPage by page name
>     - nav.html use of .Site.GetPage by page name, use of .Page.Section and .Page.Kind

This is intended. Multilingual sites identify content via file names. So all sections and files have to be named the same.

> "Fix to big margin for footer top icon" I don't understand?

This is just a note to myself :smile: 

> The owlCarousel should be implemented as well for the modal project view

:+1:. I will work on this

>  The current timezone in index page Contact section doesn't seem to work. As in #64 I'm okay with taking a vote to see if anyone actually uses this feature and wants it to remain.

:+1: This would be great.

>   Should exampleSite/resources/_gen/ be included or ignored?

I'm not sure. Bep recommends committing everything in `resources/` (See https://discourse.gohugo.io/t/hugo-0-43-released/12814/50), but I not sure. You have to commit these files every time, ons file is changed. I would suggest to add them for the moment.
 
>   Some assets don't load if baseurl includes a path, eg. https://example.org/site/

Good catch, I will fix this

**Comment by Hanzei** [comment]
My problem with your approach is, that the design changes every time you add a new project. This can confuse end users and theme users. I think the overhead is not to big. By default a three column layout is used. So the user doesn't have to worry about this.

**Comment by Hanzei** [comment]
Okay, I think I`m fine with the code. @vickylai mind giving this a round of review before  I start working on the documentation?

**Comment by Hanzei** [comment]
@vickylai I went ahead and created a draft for the documentation. Can you look at this? I`m not a native speak. Is it easy to understand? I think you way better add writing documentation.

Could you also take a look at #71.

**Comment by Hanzei** [comment]
@vickylai I went ahead and created a draft for the documentation. Can you look at this? I`m not a native speak. Is it easy to understand? I think you way better add writing documentation.

Could you also take a look at #71?

**Comment by Hanzei** [comment]
@vickylai Any updates on this PR?

**Comment by vickylai** [comment]
Overall very good. 

On single post page, I see a link in the menu to the "Lorem" part of the home page, but the link is to "/". Does this need to change to headless?

screenshot from 2018-08-21 15-51-01

I will review #71 and the documentation this week.

**Comment by Hanzei** [comment]
Thanks for your feedback!

It should link to `/#ipsum` but didnt due to a bug, which I fixed just now.

**Pull request merged**